src/main/java/com/artipie/asto/bench/BenchmarkStorage.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * The MIT License (MIT) Copyright (c) 2020-2021 artipie.com
+ * https://github.com/artipie/asto/LICENSE.txt
+ */
+package com.artipie.asto.bench;
+
+import com.artipie.asto.ArtipieIOException;
+import com.artipie.asto.Concatenation;
+import com.artipie.asto.Content;
+import com.artipie.asto.Key;
+import com.artipie.asto.OneTimePublisher;
+import com.artipie.asto.Remaining;
+import com.artipie.asto.Storage;
+import com.artipie.asto.ValueNotFoundException;
+import com.artipie.asto.ext.CompletableFutureSupport;
+import com.artipie.asto.memory.InMemoryStorage;
+import hu.akarnokd.rxjava2.interop.SingleInterop;
+import io.vavr.NotImplementedError;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentSkipListMap;
+import java.util.function.Function;
+
+/**
+ * Storage implementation for benchmarks.
+ * @since 1.1.0
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+public final class BenchmarkStorage implements Storage {
+    /**
+     * Backend storage.
+     */
+    private final InMemoryStorage backend;
+
+    /**
+     * Local storage.
+     */
+    private final Map<Key, byte[]> local;
+
+    /**
+     * Set which contains deleted keys.
+     */
+    private final Set<Key> deleted;
+
+    /**
+     * Ctor.
+     * @param backend Backend storage
+     */
+    public BenchmarkStorage(final InMemoryStorage backend) {
+        this(backend, Collections.emptyMap());
+    }
+
+    /**
+     * Ctor.
+     * @param backend Backend storage
+     * @param local Content of local storage
+     */
+    public BenchmarkStorage(final InMemoryStorage backend, final Map<Key, byte[]> local) {
+        this.backend = backend;
+        this.local = new ConcurrentSkipListMap<>(local);
+        this.deleted = ConcurrentHashMap.newKeySet();
```

Comment:
Why not `new ConcurrentSkipListSet<>`?

Reply:
@g4s8 I think that we don't need to keep order for this set, therefore usual set without sorting is more suitable here